eng/scripts/job-matrix/job-matrix-functions.modification.tests.ps1

```diff
@@ -0,0 +1,151 @@
+        $matrix.Length | Should -Be 12
+    }
+
+    It "Should combine with multiple nonSparse fields" {
+        $matrixJson = @'
+{
+    "matrix": {
+        "testField1": [ 1, 2 ],
+        "testField2": [ 1, 2 ],
+        "testField3": [ 31, 32 ],
+        "testField4": [ 41, 42 ]
+    }
+}
+'@
+        $config = GetMatrixConfigFromJson $matrixJson
+
+        $matrix = GenerateMatrix $config "all" -nonSparseParameters "testField3","testField4"
+        $matrix.Length | Should -Be 16
+
+        $matrix = GenerateMatrix $config "sparse" -nonSparseParameters "testField3","testField4"
+        $matrix.Length | Should -Be 8
+    }
+}
+
+Describe "Platform Matrix Import" -Tag "import" {
+    It "Should generate a matrix with nonSparseParameters and an imported sparse matrix" {
+        $matrixJson = @'
+{
+    "matrix": {
+        "$IMPORT": "./test-import-matrix.json",
+        "testField": [ "test1", "test2" ]
+    }
+}
+'@
+        $importConfig = GetMatrixConfigFromJson $matrixJson
+        $matrix = GenerateMatrix $importConfig "sparse" -nonSparseParameters "testField"
+
+        $matrix.Length | Should -Be 6
+
+        $matrix[0].name | Should -Be test1_foo1_bar1
+        $matrix[0].parameters.testField | Should -Be "test1"
+        $matrix[0].parameters.Foo | Should -Be "foo1"
+        $matrix[2].name | Should -Be test1_importedBaz
+        $matrix[2].parameters.testField | Should -Be "test1"
+        $matrix[2].parameters.Baz | Should -Be "importedBaz"
+        $matrix[4].name | Should -Be test2_foo2_bar2
+        $matrix[4].parameters.testField | Should -Be "test2"
+        $matrix[4].parameters.Foo | Should -Be "foo2"
+    }
+
+    It "Should generate a sparse matrix with an imported a sparse matrix" {
+        $matrixJson = @'
+{
+    "matrix": {
+        "$IMPORT": "./test-import-matrix.json",
+        "testField1": [ "test11", "test12" ],
+        "testField2": [ "test21", "test22" ]
+    }
+}
+'@
+        $importConfig = GetMatrixConfigFromJson $matrixJson
+        $matrix = GenerateMatrix $importConfig "sparse"
+
+        $matrix.Length | Should -Be 6
+
+        $matrix[0].name | Should -Be test11_test21_foo1_bar1
+        $matrix[0].parameters.testField1 | Should -Be "test11"
+        $matrix[0].parameters.testField2 | Should -Be "test21"
+        $matrix[0].parameters.Foo | Should -Be "foo1"
+        $matrix[2].name | Should -Be test11_test21_importedBaz
+        $matrix[2].parameters.testField1 | Should -Be "test11"
+        $matrix[2].parameters.testField2 | Should -Be "test21"
+        $matrix[2].parameters.Baz | Should -Be "importedBaz"
+        $matrix[4].name | Should -Be test12_test22_foo2_bar2
+        $matrix[4].parameters.testField1 | Should -Be "test12"
+        $matrix[4].parameters.testField2 | Should -Be "test22"
+        $matrix[4].parameters.Foo | Should -Be "foo2"
+    }
+
+    It "Should import a sparse matrix with import, include, and exclude" {
+        $matrixJson = @'
+{
+    "matrix": {
```

Comment:
From a testing prospective and an understanding prospective it would be good to add a test where you you are using the import and compare it to one where you didn't do the import so we can understand how things would look if we didn't import it.

Reply:
Great idea